components/automate-ui/src/app/pages/policy/add-members/policy-add-members.component.ts

```diff
@@ -305,4 +328,68 @@ export class PolicyAddMembersComponent implements OnInit, OnDestroy {
     this.addOrRemoveQueuedMember(true, member);
     this.closeModal();
   }
+
+  public showInputs(inputName: string): boolean {
+    const typeValue = this.expressionForm.get('type').value;
+    const identityValue = this.expressionForm.get('identity').value;
+
+    this.setFormLabels(typeValue);
+
+    switch (inputName) {
+      case 'identity':
+        if (typeValue === 'user' || typeValue === 'team') {
+            this.updateValidations(inputName, true);
+            return true;
+          } else {
+            this.updateValidations(inputName, false);
+            this.expressionForm.get(inputName).reset();
+            return false;
+          }
+        break;
+      case 'name':
+        if ( typeValue === 'token' || (identityValue && identityValue !== '*') ) {
```

Comment:
Prefer to use a named constant for clarity:
```suggestion
        if ( typeValue === 'token' || (identityValue && identityValue !== matchAllWildcard) ) {
```